__tests__/app.test.js

```diff
@@ -184,3 +183,72 @@ describe("ERROR:GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201: should add a new comment to the comments", () => {
+    return request(app)
+      .post(`/api/articles/1/comments`)
+      .expect(201)
+      .send({
+        username: "butter_bridge",
+        body: "This is awesome!",
+      })
+      .then(({ body: { comment } }) => {
+        expect(comment).toHaveProperty("comment_id", expect.any(Number));
+        expect(comment).toHaveProperty("body", expect.any(String));
+        expect(comment).toHaveProperty("article_id", expect.any(Number));
+        expect(comment).toHaveProperty("author", expect.any(String));
+        expect(comment).toHaveProperty("votes", expect.any(Number));
+        expect(comment).toHaveProperty("created_at", expect.any(String));
+      });
+  });
+});
+
+describe("ERROR: POST /api/articles/:article_id/comments", () => {
+  test("400: responds with an error message when passed a bad request", () => {
+    const article_id = "invalid_type";
+    return request(app)
+      .post(`/api/articles/${article_id}/comments`)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad Request");
+      });
+  });
+
+  test("404: responds with an error message when we try to post a comment to an article that doesn't exist", () => {
+    return request(app)
+      .post(`/api/articles/32435/comments`)
+      .send({
+        username: "chabalaza",
```

Comment:
For this test I'd recommend using a valid username - as if we introduce two errors in the test its difficult to be confident that you're erroring for the right reason.